[BUG] LNK4217 and LNK4286 warnings when linking //:benchmark_main

```
…
INFO: From Linking regexp_benchmark.exe:
   Creating library bazel-out\x64_windows-dbg\bin\regexp_benchmark.lib and object bazel-out\x64_windows-dbg\bin\regexp_benchmark.exp
LINK : warning LNK4217: symbol '?RegisterBenchmarkInternal@internal@benchmark@@YAPEAVBenchmark@12@PEAV312@@Z (class benchmark::internal::Benchmark * __cdecl benchmark::internal::RegisterBenchmarkInternal(class benchmark::internal::Benchmark *))' defined in 'benchmark.lib(benchmark_register.obj)' is imported by 'regexp_benchmark.obj' in function '"void __cdecl re2::`dynamic initializer for 'benchmark_uniq_100_benchmark_''(void)" (??__Ebenchmark_uniq_100_benchmark_@re2@@YAXXZ)'
LINK : warning LNK4217: symbol '?InitializeStreams@internal@benchmark@@YAHXZ (int __cdecl benchmark::internal::InitializeStreams(void))' defined in 'benchmark.lib(benchmark.obj)' is imported by 'regexp_benchmark.obj' in function '"void __cdecl benchmark::internal::`dynamic initializer for 'stream_init_anchor''(void)" (??__Estream_init_anchor@internal@benchmark@@YAXXZ)'
LINK : warning LNK4286: symbol '?InitializeStreams@internal@benchmark@@YAHXZ (int __cdecl benchmark::internal::InitializeStreams(void))' defined in 'benchmark.lib(benchmark.obj)' is imported by 'benchmark_main.lib(benchmark_main.obj)'
…
```

(The full logs are [here](https://github.com/google/re2/runs/5558969232?check_suite_focus=true).)

Oh, and I just noticed this too:

```
…
INFO: From Compiling src/benchmark_main.cc:
external/com_github_google_benchmark/src/benchmark_main.cc(18): warning C4273: 'main': inconsistent dll linkage
external/com_github_google_benchmark/src/benchmark_main.cc(17): note: see previous definition of 'main'
…
```

What's up with that? O_o

--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -55,10 +55,11 @@ cc_library(
         ":windows": [],
         "//conditions:default": posix_copts,
     }),
-    local_defines = select({
-        ":windows": ["benchmark_EXPORTS"],
-        "//conditions:default": [],
-    }),
+    # Only static linking is allowed; no .so will be produced.
+    # Using `defines` (i.e. not `local_defines`) means that no
+    # dependent rules need to bother about defining the macro.
+    linkstatic = True,
+    defines = ["BENCHMARK_STATIC_DEFINE"],
 )
 
 cc_library(

--- a/bindings/python/build_defs.bzl
+++ b/bindings/python/build_defs.bzl
@@ -14,7 +14,6 @@ def py_extension(name, srcs, hdrs = [], copts = [], features = [], deps = []):
             copts = copts,
             features = features,
             deps = deps,
-            local_defines = ["BENCHMARK_STATIC_DEFINE"],
         )
 
     return native.py_library(

